*** dot_parser.orig.py	2018-08-20 15:33:51.107742504 -0700
--- dot_parser.py	2018-08-20 15:51:23.759256197 -0700
***************
*** 1,4 ****
- # -*- coding: Latin-1 -*-
  """Graphviz's dot language parser.
  
  The dotparser parses graphviz files in dot and dot files and transforms them
--- 1,3 ----
*************** Ported by: Eric Chio <ckieric@gmail.com>
*** 14,20 ****
  __author__ = ['Michael Krause', 'Ero Carrera', 'Eric Chio']
  __license__ = 'MIT'
  
! 
  import sys
  import glob
  import pydot
--- 13,19 ----
  __author__ = ['Michael Krause', 'Ero Carrera', 'Eric Chio']
  __license__ = 'MIT'
  
! import builtins
  import sys
  import glob
  import pydot
*************** def push_top_graph_stmt(str, loc, toks):
*** 78,84 ****
      for element in toks:
      
          if( isinstance(element, (ParseResults, tuple, list)) and
!             len(element) == 1 and isinstance(element[0], str) ):
              
              element = element[0]
              
--- 77,83 ----
      for element in toks:
      
          if( isinstance(element, (ParseResults, tuple, list)) and
!             len(element) == 1 and isinstance(element[0], builtins.str) ):
              
              element = element[0]
              
*************** def push_top_graph_stmt(str, loc, toks):
*** 94,100 ****
              
              top_graphs.append( g )
              
!         elif isinstance( element, str):
              g.set_name( element )
              
          elif isinstance(element, pydot.Subgraph):
--- 93,99 ----
              
              top_graphs.append( g )
              
!         elif isinstance( element, builtins.str):
              g.set_name( element )
              
          elif isinstance(element, pydot.Subgraph):
*************** def parse_dot_data(data):
*** 508,515 ****
      
      top_graphs = list()
  
!     if data.startswith(codecs.BOM_UTF8):
!         data = data.decode( 'utf-8' )
          
      try:
      
--- 507,513 ----
      
      top_graphs = list()
  
!     data = data.decode( 'utf-8' )
          
      try:
      
*** __init__.orig.py	2018-08-20 15:30:09.000000000 -0700
--- __init__.py	2018-12-20 12:20:35.767305580 -0800
*************** class Dot(Graph):
*** 2010,2016 ****
              stderr.close()
  
              if stderr_output:
!                 stderr_output = ''.join(stderr_output)
  
          #pid, status = os.waitpid(p.pid, 0)
          status = p.wait()
--- 2010,2016 ----
              stderr.close()
  
              if stderr_output:
!                 stderr_output = b''.join(stderr_output)
  
          #pid, status = os.waitpid(p.pid, 0)
          status = p.wait()
